cyclonedx/model/bom.py

```diff
@@ -115,22 +116,30 @@ def timestamp(self, timestamp: datetime) -> None:
     #    ... # TODO since CDX1.5
 
     @property
+    @serializable.type_mapping(ToolRepositoryHelper)
     @serializable.xml_array(serializable.XmlArraySerializationType.NESTED, 'tool')
     @serializable.xml_sequence(3)
-    def tools(self) -> 'SortedSet[Tool]':
+    def tools(self) -> ToolRepository:
         """
         Tools used to create this BOM.
 
         Returns:
             `Set` of `Tool` objects.
         """
-        # TODO since CDX1.5 also supports `Component` and `Services`, not only `Tool`
-        return self._tools
+        return self._tools  # type: ignore
 
     @tools.setter
-    def tools(self, tools: Iterable[Tool]) -> None:
-        # TODO since CDX1.5 also supports `Component` and `Services`, not only `Tool`
-        self._tools = SortedSet(tools)
+    def tools(self, tools: Union[ToolRepository, Iterable[Tool]]) -> None:
```

Comment:
i dont get the conceptual idea behind this.

if `Iterable[Tool]` was a thing, the we would not need `ToolRepository.tools` at all.
And since we have `ToolRepository.tools`, we dont need `Iterable[Tool]`...


Please decide on one of the concepts and dont mix them.